README.rst

```diff
@@ -24,30 +24,86 @@ WebSockets
 .. |codecov| image:: https://codecov.io/gh/aaugustin/websockets/branch/master/graph/badge.svg
     :target: https://codecov.io/gh/aaugustin/websockets
 
+What is WebSockets?
+-------------------
 
-``websockets`` is a library for developing WebSocket servers_ and clients_ in
-Python. It implements `RFC 6455`_ and `RFC 7692`_ with a focus on correctness
-and simplicity. It passes the `Autobahn Testsuite`_.
+``websockets`` is a library for building WebSocket servers_ and clients_ in
+Python with a focus on correctness and simplicity.
 
-Built on top of Python's asynchronous I/O support introduced in `PEP 3156`_,
-it provides an API based on coroutines, making it easy to write highly
-concurrent applications.
+.. _servers: https://github.com/aaugustin/websockets/blob/master/example/server.py
+.. _clients: https://github.com/aaugustin/websockets/blob/master/example/client.py
+
+Built on top of ``asyncio``, Python's standard asynchronous I/O framework, it
+provides an elegant coroutine-based API.
+
+Here's a client that says "Hello world!":
+
+.. include:: example/hello.py
+    :code: python
+
+And here's an echo server:
+
+.. include:: example/echo.py
+    :code: python
+
+Does that sound good? `Start here`_.
+
+.. _Start here: https://websockets.readthedocs.io/en/stable/intro.html
+
+Why should I use WebSockets?
+----------------------------
+
+The development of ``websockets`` is shaped by four principles:
+
+1. **Simplicity**: all you need to understand is ``msg = await ws.recv()`` and
+   ``await ws.send(msg)``; ``websockets`` takes care of managing connections
+   so you can focus on your application.
+
+2. **Robustness**: ``websockets`` was built for production; for example it was
+   the only library to `handle backpressure correctly`_ before the issue
+   became widely known in the Python community.
 
-Installation is as simple as ``pip install websockets``.
+3. **Quality**: ``websockets`` is heavily tested. Continuous integration fails
+   under 100% branch coverage. Also it passes the industry-standard `Autobahn
+   Testsuite`_.
 
-It requires Python ≥ 3.4.
+4. **Performance**: memory use is configurable. An extension written in C
+   accelerates expensive operations. It's pre-compiled for Linux, macOS and
+   Windows and packaged in the wheel format for each system and Python version.
 
-Documentation is available on `Read the Docs`_.
+Documentation is a first class concern in the project. Head over to `Read the
+Docs`_ and see for yourself.
+
+Professional support is available if you — or your company — are so inclined.
+`Get in touch`_.
```

Comment:
I'm creating a PR rather than committing directly to master because I want to address this point. Given the conflicted relationship between open-source and feeding one's family, it may ruffle some feathers.

The general background is the whole discussion about funding open source. If you're unfamiliar, look at what Nadia Eghbal and Russell Keith-Magee said or wrote on this topic.

Without further ado:

1. This is a very low-key way to advertise my services. I would be pleasantly surprised if a single company offered to send money my way as a consequence of this sentence. I fully expect this experiment to prove that you can't make money from open source simply by offering paid support to people who already get free support — even if it isn't always timely or satisfactory.

2. If other major contributors to the library are also interested in the highly-improbable eventuality of getting a contract that way, let's turn this into a list of companies. I'm OK with listing 100% personally owned companies of anyone whose contributions prove (1) a regular involvement over a tangible period of time (2) a good understanding of the library.

FWIW my hourly rate x the time I put into websockets crossed six digits in euros or dollars some ago. There's a long way to making this a profitable venture... It's a hobby.

Reply:
FWIW, I don't see a problem with this sentence because it's your project, so you can do what you want! From your perspective, I think the main thing you'd need to worry about is turning off potential contributors. I think the offer to list other contributors is a generous one. You might want to add that to the text itself because most contributors probably won't see the comment you're making here.

Reply:
I hope it will stop being just *my* project at some point :-)

I'm going to do what you suggested, thanks.